packages/terra-functional-testing/src/services/wdio-terra-service.js

```diff
@@ -2,18 +2,68 @@
 const expect = require('expect');
 const { accessibility } = require('../commands/validates');
 const { toBeAccessible } = require('../commands/expect');
+const { getViewports, describeViewports, setViewport } = require('../commands/viewport-helpers');
+
+const hideInputCaret = require('../commands/hide-input-caret');
 
 class TerraService {
   /**
    * Service hook executed prior to test execution.
    * Initializes the Terra Service's custom commands.
    */
-  before() {
+  before(capabilities) {
+    // Add the Jest expect module the use the Jest matchers.
     global.expect = expect;
     global.expect.extend({ toBeAccessible });
 
-    global.Terra = {};
-    global.Terra.validates = { accessibility };
+    // Add a Terra global with access to Mocha-Chai test helpers.
+    global.Terra = {
+      validates: { accessibility },
+
+      // viewports provides access Terra's list of test viewports.
+      viewports: getViewports,
+
+      // describeViewports provides a custom describe block for looping test viewports.
+      describeViewports,
+
+      // hideInputCaret hides the blinking input caret that appears in inputs or editable text areas.
+      hideInputCaret,
+    };
+
+    // IE driver takes longer to be ready for browser interactions.
+    if (capabilities.browserName === 'internet explorer') {
+      global.browser.$('body').waitForExist({
+        timeout: global.browser.config.waitforTimeout,
+        interval: 100,
+      });
+    }
+
+    // Extract the Terra service options from the global browser object.
+    const [, options = {}] = global.browser.options.services.find(([service]) => (
+      typeof service === 'function' && service.name === 'TerraService'
+    ));
```

Comment:
We should add a constructor and set the formFactor like:
```
class TerraService {
  constructor(options) {
    this.formFactor = options.formFactor;
  }
  ..
   setViewport(this.formFactor);

```
we shouldn't have to pull this config from the global here

Reply:
Of my latests comments, this is the main one that I feel really needs to be addressed.

Reply:
Would this mean that anywhere that initializes TerraService would need to first get the options? My thought was to abstract that and get the service option here.

Reply:
The TerraService is initialized with the options, we can save those and then use them directly rather than looking for them on the global config.

Here's an example of the WDIO sauce service:

https://github.com/webdriverio/webdriverio/blob/master/packages/wdio-sauce-service/src/service.js#L10-L14

The utility functions need to search for the global options since they don't have a reference to `this` for the TerraService (although we might be able to bind them with the options).

Reply:
Yes, but we are initializing this in the wdio configuration

Reply:
After running some tests and saw that the service options is passed to the service and already available. So yes, getting it in the constructor is more efficient. https://github.com/cerner/terra-toolkit/pull/438/commits/bf231c014c5f0e7bb7c11400e6fe1b314f146a41